source/PlayerInfo.cpp

```diff
@@ -1945,6 +1945,20 @@ void PlayerInfo::AddSpecialLog(const string &type, const string &name, const str
 
 
 
+void PlayerInfo::RemoveSpecialLog(const std::string &type, const std::string &name)
+{
+	specialLogs[type].erase(specialLogs[type].find(name));
```

Comment:
From: https://en.cppreference.com/w/cpp/container/map/erase
> The iterator pos must be valid and dereferenceable. Thus the [end()](https://en.cppreference.com/w/cpp/container/map/end) iterator (which is valid, but is not dereferenceable) cannot be used as a value for pos.

So, if the `std::map` returned by `specialLogs[type]` does not contain a key equivalent to `name`, this will be bad, because it'll attempt to `erase` the `end()` iterator.